core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -204,35 +209,43 @@ private[spark] class CoarseMesosSchedulerBackend(
 
       for (offer <- offers) {
         val slaveId = offer.getSlaveId.toString
-        val mem = getResource(offer.getResourcesList, "mem")
-        val cpus = getResource(offer.getResourcesList, "cpus").toInt
-        if (totalCoresAcquired < maxCores &&
-            mem >= MemoryUtils.calculateTotalMemory(sc) &&
-            cpus >= 1 &&
+        var totalMem = getResource(offer.getResourcesList, "mem")
+        var totalCpus = getResource(offer.getResourcesList, "cpus").toInt
```

Comment:
also, `totalMem` and `totalCpus` aren't exactly correct descriptions of what these variables represent. I think better names are `offerMem` and `offerCores` (i.e. these are the resources given to us by the offer)